src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copiedState = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(copiedState, action.extraData);
+        break;
+      case 'removeProperties':
+        removeProperties(copiedState, action.keysToRemove);
+        break;
+      case 'clear':
+        clear(copiedState);
+        break;
```

Comment:
3. [CheckList] switch/case should always have default case for error handling.
there have to be something like this 
default:
        throw new Error('action.type not found');